test/unittests.jl

```diff
@@ -10,9 +11,9 @@ using Test
 
 @testset "Basic utilities" begin
 
-    for T in [DEFAULT_VALUE_TYPE, Float16, Float32, Float64], R in [DEFAULT_DIM_BASE_TYPE, Rational{Int16}, Rational{Int32}, SimpleRatio{Int}, SimpleRatio{SafeInt16}]
+    for Q in [Quantity, GenericQuantity], T in [DEFAULT_VALUE_TYPE, Float16, Float32, Float64], R in [DEFAULT_DIM_BASE_TYPE, Rational{Int16}, Rational{Int32}, SimpleRatio{Int}, SimpleRatio{SafeInt16}]
```

Comment:
@gaurav-arya much of `AbstractUnionQuantity` should be covered by this generalization. Maybe I could do it for all the tests but since they are identical I guess we don't need too much... Just need to be careful for things that are not generalized to AbstractUnionQuantity